clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1179,6 +1186,50 @@ protected String getConfigValue(ConfigKey key, String headerName) {
         }
     }
 
+    private static String niceMemoryUnits(ConfigKey key) {
+        final long defaultValueBytes = ((Number) key.defaultValue).longValue();
+        long defaultValue = defaultValueBytes;
+        int i = 0;
+        while (i < 5) {
+            if (defaultValue % 1024 != 0) {
+                break;
+            }
+            defaultValue /= 1024;
+            i++;
+        }
+        switch (i) {
+            case 1:
+                return " (=" + defaultValue + "KiB)";
+            case 2:
+                return " (=" + defaultValue + "MiB)";
+            case 3:
+                return " (=" + defaultValue + "GiB)";
+            case 4:
+                return " (=" + defaultValue + "TiB)";
+            default:
+                return "";
+        }
+    }
+
+    private static String niceTimeUnits(ConfigKey key) {
+        final long defaultValueBytes = ((Number) key.defaultValue).longValue();
+        long defaultValue = defaultValueBytes;
+        if (defaultValueBytes % 1000L * 60 * 60 * 24 == 0) {
```

Comment:
I wonder if using Durations would make this more readable. For example `Duration.ofDays(1).toMillis()` instead of `1000L * 60 * 60 * 24`. WDYT?